### ISSUE TITLE
Reducing number of variants in results

### ISSUE THREAD

**Issue by 11Dmitriy11** [Ensembl/postgap#179]
Hello! Could you please help me to understand one moment about final quantity of variants in resulting file?
Initially I had 90 variants with p-value < threshold. Seems that each variant was analysed but there were only 20 variants in a resulting file. Also I found this statement ' Searching for cis-regulatory data on 20 SNPs in all databases'. However I don't understand which filters were applyed to reduce number of variants from 90 to 20

04/28/2022 11:39:50 PM - root - INFO - Found 90 gwas associations with a pvalue of 0.01 or less.
04/28/2022 11:39:50 PM - root - INFO - Done searching for GWAS SNPs. Found 90 gwas associations.
04/28/2022 11:39:50 PM - root - INFO - Found 90 unique GWAS SNPs associated to diseases () or EFO IDs () in all databases
04/28/2022 11:39:50 PM - root - INFO - Found 90 GWAS SNPs associated to diseases () or EFO IDs () after p-value filter (0.010000)
04/28/2022 11:39:51 PM - root - WARNING - Waiting for http://grch37.rest.ensembl.org/variation/Human?content-type=application/json
04/28/2022 11:39:57 PM - root - INFO - Found 90 locations from 90 GWAS SNPs
04/28/2022 11:39:57 PM - root - INFO - Found 1 SNPs in the vicinity of rs5757042
04/28/2022 11:39:57 PM - root - INFO - Found 5 SNPs in the vicinity of rs5757043
04/28/2022 11:39:57 PM - root - INFO - Found 1 SNPs in the vicinity of rs78084712
04/28/2022 11:39:57 PM - root - INFO - Found 5 SNPs in the vicinity of rs2281108
04/28/2022 11:39:57 PM - root - INFO - Found 5 SNPs in the vicinity of rs3788532
04/28/2022 11:39:57 PM - root - INFO - Found 2 SNPs in the vicinity of rs3788533
04/28/2022 11:39:57 PM - root - INFO - Found 20 SNPs in the vicinity of rs2267377
04/28/2022 11:39:57 PM - root - INFO - Found 1 SNPs in the vicinity of rs115343378
04/28/2022 11:39:58 PM - root - INFO - Found 14 SNPs in the vicinity of rs11705000
04/28/2022 11:39:58 PM - root - INFO - Found 5 SNPs in the vicinity of rs2008720
04/28/2022 11:39:58 PM - root - INFO - Found 1 SNPs in the vicinity of rs35526737
04/28/2022 11:39:58 PM - root - INFO - Found 20 SNPs in the vicinity of rs5750571
04/28/2022 11:39:58 PM - root - INFO - Found 5 SNPs in the vicinity of rs115082049
04/28/2022 11:39:58 PM - root - INFO - Found 1 SNPs in the vicinity of rs9622722
04/28/2022 11:39:58 PM - root - INFO - Found 1 SNPs in the vicinity of rs146162374
04/28/2022 11:39:58 PM - root - INFO - Found 1 SNPs in the vicinity of rs536760759
04/28/2022 11:39:58 PM - root - INFO - Found 5 SNPs in the vicinity of rs113651082
04/28/2022 11:39:58 PM - root - INFO - Found 2 SNPs in the vicinity of rs6007506
04/28/2022 11:39:58 PM - root - INFO - Found 11 SNPs in the vicinity of rs5770489
04/28/2022 11:39:58 PM - root - INFO - Found 11 SNPs in the vicinity of rs5770488
04/28/2022 11:39:58 PM - root - INFO - Found 20 SNPs in the vicinity of rs7290372
04/28/2022 11:39:58 PM - root - INFO - Found 4 SNPs in the vicinity of rs1292811
04/28/2022 11:39:58 PM - root - INFO - Found 20 SNPs in the vicinity of rs7289126
04/28/2022 11:39:58 PM - root - INFO - Found 69 SNPs in the vicinity of rs6001047
04/28/2022 11:39:59 PM - root - INFO - Found 69 SNPs in the vicinity of rs6001046
04/28/2022 11:39:59 PM - root - INFO - Found 5 SNPs in the vicinity of rs2235347
04/28/2022 11:39:59 PM - root - INFO - Found 14 SNPs in the vicinity of rs11703294
04/28/2022 11:39:59 PM - root - INFO - Found 27 SNPs in the vicinity of rs2294196
04/28/2022 11:39:59 PM - root - INFO - Found 1 SNPs in the vicinity of rs146092440
04/28/2022 11:39:59 PM - root - INFO - Found 20 SNPs in the vicinity of rs2267378
04/28/2022 11:39:59 PM - root - INFO - Found 20 SNPs in the vicinity of rs2284072
04/28/2022 11:39:59 PM - root - INFO - Found 4 SNPs in the vicinity of rs5750620
04/28/2022 11:39:59 PM - root - INFO - Found 19 SNPs in the vicinity of rs7290309
04/28/2022 11:39:59 PM - root - INFO - Found 5 SNPs in the vicinity of rs5750591
04/28/2022 11:39:59 PM - root - INFO - Found 69 SNPs in the vicinity of rs5756959
04/28/2022 11:39:59 PM - root - INFO - Found 69 SNPs in the vicinity of rs5756958
04/28/2022 11:39:59 PM - root - INFO - Found 69 SNPs in the vicinity of rs5756956
04/28/2022 11:39:59 PM - root - INFO - Found 1 SNPs in the vicinity of rs79966207
04/28/2022 11:39:59 PM - root - INFO - Found 19 SNPs in the vicinity of rs8142686
04/28/2022 11:39:59 PM - root - INFO - Found 1 SNPs in the vicinity of rs188926108
04/28/2022 11:40:00 PM - root - INFO - Found 1 SNPs in the vicinity of rs147296046
04/28/2022 11:40:00 PM - root - INFO - Found 4 SNPs in the vicinity of rs3827354
04/28/2022 11:40:00 PM - root - INFO - Found 69 SNPs in the vicinity of rs34771206
04/28/2022 11:40:00 PM - root - INFO - Found 1 SNPs in the vicinity of rs555693482
04/28/2022 11:40:00 PM - root - INFO - Found 20 SNPs in the vicinity of rs4820329
04/28/2022 11:40:00 PM - root - INFO - Found 20 SNPs in the vicinity of rs4820328
04/28/2022 11:40:00 PM - root - INFO - Found 14 SNPs in the vicinity of rs11704832
04/28/2022 11:40:00 PM - root - INFO - Found 69 SNPs in the vicinity of rs4821757
04/28/2022 11:40:00 PM - root - INFO - Found 69 SNPs in the vicinity of rs4821758
04/28/2022 11:40:00 PM - root - INFO - Found 1 SNPs in the vicinity of rs2413508
04/28/2022 11:40:00 PM - root - INFO - Found 1 SNPs in the vicinity of rs59650998
04/28/2022 11:40:00 PM - root - INFO - Found 14 SNPs in the vicinity of rs11705041
04/28/2022 11:40:00 PM - root - INFO - Found 1 SNPs in the vicinity of rs11089862
04/28/2022 11:40:00 PM - root - INFO - Found 2 SNPs in the vicinity of rs77551271
04/28/2022 11:40:00 PM - root - INFO - Found 1 SNPs in the vicinity of rs1055658
04/28/2022 11:40:00 PM - root - INFO - Found 1 SNPs in the vicinity of rs28470336
04/28/2022 11:40:00 PM - root - INFO - Found 1 SNPs in the vicinity of rs6001030
04/28/2022 11:40:01 PM - root - INFO - Found 20 SNPs in the vicinity of rs2267380
04/28/2022 11:40:01 PM - root - INFO - Found 20 SNPs in the vicinity of rs2267381
04/28/2022 11:40:01 PM - root - INFO - Found 69 SNPs in the vicinity of rs5750559
04/28/2022 11:40:01 PM - root - INFO - Found 20 SNPs in the vicinity of rs5995558
04/28/2022 11:40:01 PM - root - INFO - Found 20 SNPs in the vicinity of rs5995554
04/28/2022 11:40:01 PM - root - INFO - Found 1 SNPs in the vicinity of rs1059804
04/28/2022 11:40:01 PM - root - INFO - Found 5 SNPs in the vicinity of rs9616648
04/28/2022 11:40:01 PM - root - INFO - Found 1 SNPs in the vicinity of rs12158303
04/28/2022 11:40:01 PM - root - INFO - Found 4 SNPs in the vicinity of rs56190056
04/28/2022 11:40:01 PM - root - INFO - Found 1 SNPs in the vicinity of rs187478773
04/28/2022 11:40:01 PM - root - INFO - Found 2 SNPs in the vicinity of rs139814298
04/28/2022 11:40:01 PM - root - INFO - Found 1 SNPs in the vicinity of rs149511036
04/28/2022 11:40:01 PM - root - INFO - Found 14 SNPs in the vicinity of rs11703323
04/28/2022 11:40:01 PM - root - INFO - Found 5 SNPs in the vicinity of rs5756922
04/28/2022 11:40:01 PM - root - INFO - Found 2 SNPs in the vicinity of rs6001008
04/28/2022 11:40:01 PM - root - INFO - Found 5 SNPs in the vicinity of rs9680560
04/28/2022 11:40:02 PM - root - INFO - Found 5 SNPs in the vicinity of rs2076118
04/28/2022 11:40:02 PM - root - INFO - Found 3 SNPs in the vicinity of rs117265627
04/28/2022 11:40:02 PM - root - INFO - Found 1 SNPs in the vicinity of rs190721121
04/28/2022 11:40:02 PM - root - INFO - Found 15 SNPs in the vicinity of rs8135333
04/28/2022 11:40:02 PM - root - INFO - Found 1 SNPs in the vicinity of rs143628410
04/28/2022 11:40:02 PM - root - INFO - Found 1 SNPs in the vicinity of rs11090826
04/28/2022 11:40:02 PM - root - INFO - Found 1 SNPs in the vicinity of rs5771990
04/28/2022 11:40:02 PM - root - INFO - Found 1 SNPs in the vicinity of rs189673589
04/28/2022 11:40:02 PM - root - INFO - Found 20 SNPs in the vicinity of rs5756984
04/28/2022 11:40:02 PM - root - INFO - Found 20 SNPs in the vicinity of rs5756986
04/28/2022 11:40:02 PM - root - INFO - Found 14 SNPs in the vicinity of rs78538621
04/28/2022 11:40:02 PM - root - INFO - Found 20 SNPs in the vicinity of rs2092076
04/28/2022 11:40:02 PM - root - INFO - Found 2 SNPs in the vicinity of rs5766565
04/28/2022 11:40:02 PM - root - INFO - Found 20 SNPs in the vicinity of rs6001063
04/28/2022 11:40:02 PM - root - INFO - Found 69 SNPs in the vicinity of rs5756963
04/28/2022 11:40:02 PM - root - INFO - Found 12 SNPs in the vicinity of rs16993718
04/28/2022 11:40:03 PM - root - INFO - Found 1 SNPs in the vicinity of rs542951282
04/28/2022 11:40:03 PM - root - INFO - Found 1 clusters from 90 GWAS SNP locations
04/28/2022 11:40:03 PM - root - INFO - Searching for cis-regulatory data on 20 SNPs in all databases

**Comment by dzerbino** [comment]
Hello @11Dmitriy11 ,

In the first step, Postgap found 90 variants:
```
04/28/2022 11:39:57 PM - root - INFO - Found 90 locations from 90 GWAS SNPs
```

For each of these 90 GWAS SNPs it then searches for their LD neighbours, creating a cluster around each SNP:
```
04/28/2022 11:39:57 PM - root - INFO - Found 1 SNPs in the vicinity of rs5757042
04/28/2022 11:39:57 PM - root - INFO - Found 5 SNPs in the vicinity of rs5757043
04/28/2022 11:39:57 PM - root - INFO - Found 1 SNPs in the vicinity of rs78084712
04/28/2022 11:39:57 PM - root - INFO - Found 5 SNPs in the vicinity of rs2281108
04/28/2022 11:39:57 PM - root - INFO - Found 5 SNPs in the vicinity of rs3788532
[...]
```

The overlapping clusters are first merged then the small ones (<10 SNPs) are removed. 

The script then goes through the clusters 1 by 1 and perform finemapping on each. As it were, the first cluster it examined contained 20 SNPs:
```
04/28/2022 11:40:03 PM - root - INFO - Searching for cis-regulatory data on 20 SNPs in all databases
```

In conclusion, there are a bunch of filters, and some SNPs will have been dropped, but the numbers 90 and 20 cannot be compared directly.

Hope this makes sense,

Daniel

**Comment by 11Dmitriy11** [comment]
Thank you, this explains the origin of observed problem. Is there a way to skip the fine-mapping step and force the interpretation of all 90 SNPs?

**Comment by dzerbino** [comment]
Hello @11Dmitriy11 ,

Yes, instead of summary stats you can provide the rsID of a variant (one by one I am afraid) with the `--rsID` option.

Admittedly I have not tested this option in a while, so it may not work 100%.

Cheers,

Daniel
